### PR TITLE
Piping changes 2: electric bogaloo (Guild buff and nerf)

### PR DESCRIPTION
**Pull request by Dissentbay**
## About The Pull Request
<details>
<summary>
	#4406 , but toned down since the issue was either in datums or how was interacting with ZAS. local server restarted 15 times, no issues and little to no performance enhancement noticed in server start
</summary>
<hr>
code obtained from Eris and adapted to this server by me

	
<hr>
</details>

## Changelog
:cl:
refactor: all pipes codes (minus meters), TEG circulators
refactor: heater/cooler code
Guild alert: supermatter is again fickle, test the waters before diving or you will have a bad time.

/:cl:

### CODE REVIEW

**Comment by Dissentbay** [comment]
and always the same error in windows build because of something in tgui that refuses to elaborate 😔

File: code/ATMOSPHERICS/components/binary_devices/passive_gate.dm [hunk 1]
```diff
@@ -41,17 +41,6 @@
 		add_underlay(T, node1, turn(dir, 180))
 		add_underlay(T, node2, dir)
 
-/obj/machinery/atmospherics/binary/passive_gate/on
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/components/binary_devices/passive_gate.dm:44]: dont remove this

File: code/ATMOSPHERICS/components/binary_devices/pipeturbine.dm [hunk 2]
```diff
@@ -73,15 +73,15 @@
 			network2.update = 1
 
 	update_icon()
-		cut_overlays()
+		overlays.Cut()
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/components/binary_devices/pipeturbine.dm:76]: every single one of these is going to brake the icons

File: code/ATMOSPHERICS/components/binary_devices/passive_gate.dm (hunk 1 above)
Comment (Dissentbay) [reply code/ATMOSPHERICS/components/binary_devices/passive_gate.dm]: unremoved

File: code/ATMOSPHERICS/components/binary_devices/pipeturbine.dm (hunk 2 above)
Comment (Dissentbay) [reply code/ATMOSPHERICS/components/binary_devices/pipeturbine.dm]: changed as many as I've found

File: code/ATMOSPHERICS/components/binary_devices/pipeturbine.dm [hunk 3]
```diff
@@ -73,15 +73,15 @@
...
 		if (dP > 10)
-			add_overlay(image('icons/obj/pipeturbine.dmi', "moto-turb"))
+			overlays += image('icons/obj/pipeturbine.dmi', "moto-turb")
```
[the remark this answers is not in the document]
Comment (Dissentbay) [reply code/ATMOSPHERICS/components/binary_devices/pipeturbine.dm]: also changed

File: code/ATMOSPHERICS/components/unary/vent_scrubber.dm [hunk 4]
```diff
@@ -205,28 +161,14 @@
 			var/transfer_moles = min(environment.total_moles, environment.total_moles*MAX_SCRUBBER_FLOWRATE/environment.volume)
 			//group_multiplier gets divided out here
 			power_draw += scrub_gas(src, scrubbing_gas, environment, air_contents, transfer_moles, power_rating)
-			if(debug)
```
[the remark this answers is not in the document]
Comment (Dissentbay) [reply code/ATMOSPHERICS/components/unary/vent_scrubber.dm:208]: kept this one

[the remark this answers is not in the document]
Comment (Dissentbay) [reply code/ATMOSPHERICS/components/unary/vent_scrubber.dm]: correction: kept all if(debug) with the stuff under it included in this file

**Comment by Dissentbay** [comment]
now to keep not understanding these TGUI errors, any help is welcome

**Comment by Dissentbay** [comment]
hold on, those changes made it harder to start, SOMEHOW

**Comment by Dissentbay** [comment]
yep, the changes made it the server to stall, time to comment out each of them to find the fuck up, since before it managed to restart the server 15 times back to back just fine

**Comment by Dissentbay** [comment]
in my machine the if(debug) part was causing the dream daemon to stall before loading anything

**Comment by Dissentbay** [comment]
as it is, is now working on my machine as intended. and not stalling anymore

File: code/ATMOSPHERICS/components/omni_devices/omni_base.dm [hunk 5]
```diff
@@ -48,11 +48,11 @@
 
 /obj/machinery/atmospherics/omni/update_icon()
 	if(stat & NOPOWER)
-		copy_overlays(overlays_off, TRUE)
+		overlays = overlays_off
```
Comment (Trilbyspaceclone) [inline code/ATMOSPHERICS/components/omni_devices/omni_base.dm:51]: These ones are still using the broken systme of overlays

Comment (Dissentbay) [reply code/ATMOSPHERICS/components/omni_devices/omni_base.dm]: fixed now, all overlays on file equal to the original

**Comment by Dissentbay** [comment]
all working, except the same tgui error since many prs ago

**Comment by Dissentbay** [comment]
was forced to change omni_base.dm to the other overlay type, because the system we use currently is incompatible (literally no overlay happens on this file if you try any of the fancy atom handling of overlays instead of the crude ones)

**Pull request merged**